library/std/src/sys/pal/unix/process/process_unix.rs

```diff
@@ -640,6 +702,47 @@ impl Command {
             #[cfg(target_os = "nto")]
             let spawn_fn = retrying_libc_posix_spawnp;
 
+            #[cfg(target_os = "linux")]
+            if self.get_create_pidfd() && PIDFD_SUPPORTED.load(Ordering::Relaxed) == SPAWN {
+                let mut pidfd: libc::c_int = -1;
+                let spawn_res = pidfd_spawnp.get().unwrap()(
+                    &mut pidfd,
+                    self.get_program_cstr().as_ptr(),
+                    file_actions.0.as_ptr(),
+                    attrs.0.as_ptr(),
+                    self.get_argv().as_ptr() as *const _,
+                    envp as *const _,
+                );
```

Comment:
Ah, I see why they didn't add another argument to `pidfd_spawnp`: this way it is "exactly the same" arg/ret as `posix_spawnp` instead of adding another one. They are 1:1 except for "you get a pidfd, not a pid", right?